base/REPLCompletions.jl

```diff
@@ -232,11 +232,11 @@ function find_start_brace(s::AbstractString)
                 in_back_ticks = true
             end
         else
-            if !in_back_ticks && !in_double_quotes && c == '\''
+            if !in_back_ticks && !in_double_quotes && c == '\'' && next(r, i)[1]!='\\'
```

Comment:
I have adjusted the PR to follow your comment. The current use of the function ensures a bounds errror wouldn't occur, but I have added the `done` condition to make the function safe.